src/types/any.rs

```diff
@@ -143,6 +157,39 @@ impl PyAny {
     /// else:
     ///     raise TypeError("PyAny::compare(): All comparisons returned false")
     /// ```
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use pyo3::prelude::*;
+    /// use pyo3::types::PyFloat;
+    /// use std::cmp::Ordering;
+    ///
+    /// # fn main() -> PyResult<()> {
+    /// Python::with_gil(|py| -> PyResult<()> {
+    ///     let a = PyFloat::new(py, 0_f64);
+    ///     let b = PyFloat::new(py, 42_f64);
+    ///     assert_eq!(a.compare(b)?, Ordering::Less);
+    ///     Ok(())
+    /// })?;
+    /// # Ok(())}
+    /// ```
+    ///
+    /// It will return `PyErr` for values that cannot be compared:
```

Comment:
With floats, this is due to neither of the three comparisons returning True, which is already shown in the equivalent code above.

Can you make an example that involves values that actually cannot be compare, like `int < str`?

Reply:
👍